pkg/graph/resolvers/system_intake_status_admin_test.go

```diff
@@ -0,0 +1,387 @@
+package resolvers
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+type testSystemIntakeAdminStatusType struct {
+	testCase       string
+	intake         models.SystemIntake
+	expectedStatus models.SystemIntakeStatusAdmin
+	expectError    bool
+}
+
+// TestSystemIntakeStatusAdminGet tests the possible permutations for the systemIntakeStatusAdminGet method.
+// It doesn't use the testify suit, as this is not dependant on anything besides the resolver.
+func TestSystemIntakeStatusAdminGet(t *testing.T) {
+	yesterday := time.Now().Add(time.Hour * -24)
+	tomorrow := time.Now().Add(time.Hour * 24)
+	intakeFormTests := []testSystemIntakeAdminStatusType{
+		{
+			testCase: "Request form not started",
+			intake: models.SystemIntake{
+				Step:             models.SystemIntakeStepINITIALFORM,
+				RequestFormState: models.SIRFSNotStarted,
+				State:            models.SystemIntakeStateOPEN,
+			},
+			expectedStatus: models.SISAInitialRequestFormInProgress,
+			expectError:    false,
+		},
+		{
+			testCase: "Request form in progress",
+			intake: models.SystemIntake{
+				Step:             models.SystemIntakeStepINITIALFORM,
+				RequestFormState: models.SIRFSInProgress,
+				State:            models.SystemIntakeStateOPEN,
+			},
+			expectedStatus: models.SISAInitialRequestFormInProgress,
+			expectError:    false,
+		},
+		{
+			testCase: "Request form edits requested",
+			intake: models.SystemIntake{
+				Step:             models.SystemIntakeStepINITIALFORM,
+				RequestFormState: models.SIRFSEditsRequested,
+				State:            models.SystemIntakeStateOPEN,
+			},
+			expectedStatus: models.SISAInitialRequestFormInProgress,
+			expectError:    false,
+		},
+		{
+			testCase: "Request form submitted",
+			intake: models.SystemIntake{
+				Step:             models.SystemIntakeStepINITIALFORM,
+				RequestFormState: models.SIRFSSubmitted,
+				State:            models.SystemIntakeStateOPEN,
+			},
+			expectedStatus: models.SISAInitialRequestFormSubmitted,
+			expectError:    false,
+		},
+		{
+			testCase: "Request form closed",
```

Comment:
Just to check - this is testing the case where an intake is closed while in the request form step, correct?

Reply:
That is correct